Enable Learning Mode blocks to be configured by theme.json

gabrielcaires opened the pull request:
Depends on https://github.com/Automattic/sensei/pull/6066

### Changes proposed in this Pull Request
* Update some learning mode blocks to enable get configuration from block-based themes.


### Context 
As part of the new sensei theme initiative,  we need to have the blocks getting more style configurations  (typography, spacing... ) from theme.json files and configurable by the editor.


### Testing instructions
* Switch to the branch
* run `npm run start`
* Go to the `Lesson (Learning Mode - Default)` template editor
* Check if any block is broken
* Check is possible to change the font family, font weight, font style, and decoration on the editor.
*  Make a change, save and publish and check if the rendered version is following the same style applied on the editor:

Affected Blocks:

* Course Navigation
* Course-theme-course-progress-counter (information on the header)
* Course-theme-lesson-module (Module title above the lesson name)
* Course Title (Header)
* Exit Course Link (Header)
*  Progress Bar (Above bellow the header)  <--- This one change is related to the received background colors from theme.json: 


You can also make changes by adding the following instructions using the `theme.json` on your active block based theme:


```
                   "sensei-lms/course-title" : {
				"typography": {
					"textTransform": "uppercase",
					"fontFamily": "var(--wp--preset--font-family--league-gothic)",
					"fontSize": "var(--wp--preset--font-size--medium)"
				}
			},
			
			"sensei-lms/course-theme-course-progress-bar": {
				"color": {
					"background": "#ffffff"
				}
			},
			"sensei-lms/course-theme-course-progress-counter" : {
				"typography": {
					"fontFamily": "var(--wp--preset--font-family--system)",
					"fontSize": "var(--wp--preset--font-size--xx-small)",
					"letterSpacing": "0.02em"
				}
			},
			"sensei-lms/exit-course" : {
				"typography": {
					"fontFamily": "var(--wp--preset--font-family--system)",
					"fontSize": "var(--wp--preset--font-size--xx-small)",
					"textDecoration": "underline",
					"letterSpacing": "0.02em"
				}
			},
 

```

gabrielcaires committed:
Add support to customize font family via theme json

gabrielcaires committed:
Update the course navigation block to support customizations via theme.json

gabrielcaires committed:
Update the course progress bar block to support customizations via theme.json

gabrielcaires committed:
Update lessons block

gabrielcaires committed:
Fix unit test

donnapep commented:
I tested and am able to change the typography settings for those blocks still (except for font size of the sidebar, but that looks like it didn't work before either). I also manually applied the snippet of code to `theme.json` of the Sensei theme while testing.

There are a couple of things that look broken now. I'm not sure if any of these things are expected or not, but I'll document them regardless.

The items in the sidebar are misaligned with the icons:

![Screenshot 2022-11-05 at 5 22 08 PM](https://user-images.githubusercontent.com/1190420/200141812-6a3697af-36c2-4d95-a650-f8b7f14d8cb9.jpg)

The pagination and Complete Lesson buttons are styled differently in the editor:

_Before_

![Screenshot 2022-11-05 at 5 04 37 PM](https://user-images.githubusercontent.com/1190420/200141305-db1786c4-fd0b-4f7f-9ee4-899ae3ae1a45.jpg)

_After_

![Screenshot 2022-11-05 at 5 04 30 PM](https://user-images.githubusercontent.com/1190420/200141303-ef8bc76c-f9b6-48f9-8cca-527d57bf571c.jpg)

When I tried it on Astra, the progress bar disappears altogether:

_Before_

![Screenshot 2022-11-05 at 5 19 47 PM](https://user-images.githubusercontent.com/1190420/200141754-02e052bf-78ee-405b-9635-27ffcb6eff68.jpg)

_After_

![Screenshot 2022-11-05 at 5 20 28 PM](https://user-images.githubusercontent.com/1190420/200141767-402ea05c-c710-40d5-b62a-96551b345c1f.jpg)

gabrielcaires committed:
Merge branch 'add/support-to-customize-learnning-mode-via-theme.json' into update/upgrade-learning-mode-blocks

gabrielcaires committed:
Fix progress bar background

gabrielcaires commented:
@donnapep 

> When I tried it on Astra, the progress bar disappears altogether:

Fixed here [Fix progress bar background](https://github.com/Automattic/sensei/pull/6067/commits/058b096104217141f5b915bc0aa9619f9a6deb13)

> The pagination and Complete Lesson buttons are styled differently in the editor:

The learning mode was not loading the compat css file, I fixed on the PR and merge with this one.
[2d72951](https://github.com/Automattic/sensei/pull/6066/commits/2d729518035bb1214aec5c29b86de28de3e55489)

yscik commented on assets/course-theme/blocks/lesson-blocks/course-title.block.json:
```diff
@@ -11,7 +12,16 @@
     },
     "typography": {
```
There are a couple of CSS files for these blocks (`course-title.scss`, `lesson-module.scss` etc), that contain opposing rules, including some `!important` flags. Should those, or parts of them, be also moved into `learning-mode-compat.scss` instead? (Perhaps as part of the opinionated styles PR).

Note that the block styles are currently bundled in `sensei-theme-blocks.scss`, but ideally everything staying there should be theme-agnostic. (For example the progress bar just sets up structure, but course title and lesson module are entirely opinionated styling.)

Imran92 replied:
Thanks @yscik , I've moved some styles according to your suggestion [here](https://github.com/Automattic/sensei/pull/6066/commits/483c7e49cd9954076745c3eebbea5d6f92cbb650) in the opinionated styling PR

Commit:
Merge branch 'add/support-to-customize-learnning-mode-via-theme.json' into update/upgrade-learning-mode-blocks

Commit:
Merge branch 'add/support-to-customize-learnning-mode-via-theme.json' into update/upgrade-learning-mode-blocks

Commit:
Merge branch 'add/support-to-customize-learnning-mode-via-theme.json' into update/upgrade-learning-mode-blocks

Commit:
Merge branch 'add/support-to-customize-learnning-mode-via-theme.json' into update/upgrade-learning-mode-blocks

gabrielcaires committed:
Merge branch 'trunk' into update/upgrade-learning-mode-blocks

gabrielcaires commented:
@Imran92 Tested ✅ 
Could you please approve?

Imran92 reviewed (dismissed):
Done

gabrielcaires committed:
Skip block-patterns directory

m1r0 commented on includes/blocks/class-sensei-block-take-course.php:
```diff
@@ -74,8 +74,8 @@ public function render_take_course_block( $attributes, $content ): string {
-
-		return ! empty( $html ) ? '<div class="sensei-block-wrapper sensei-cta">' . $html . '</div>' : '';
+		$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => 'sensei-block-wrapper sensei-cta' ) );
+		return ! empty( $html ) ? '<div' . $wrapper_attributes . '>' . $html . '</div>' : '';
```
I think there is a missing space after the `<div` which is causing some Course List issues.

![image](https://user-images.githubusercontent.com/1612178/203556794-955be207-4bdd-4982-8908-16254ba02fd4.png)

gabrielcaires replied:
I fixed it https://github.com/Automattic/sensei/pull/6163/files#diff-b981e2bbec3c37011de89cb5a29cb17879558d14f81a44e59e670a90e17f67caR78
